application/classes/Controller/Pages.php

```diff
@@ -2,31 +2,233 @@
 
 class Controller_Pages extends Controller_Base_preDispatch
 {
-    const TYPE_PAGE = 1;
-    const TYPE_NEWS = 2;
-    const TYPE_BLOG = 3;
 
-    public function action_page()
+
+    public function action_show_page()
     {
         $id = $this->request->param('id');
         $uri = $this->request->param('uri');
 
-        $page = $this->methods->getPage( $id , $uri );
 
-        if ( $page ) {
 
-            $this->view['page']  = $page;
-            $this->view['files'] = $this->methods->getPageFiles( $page['id'] );
+        $page = Model_Page::get($id);
+        if ($page->title)
+        {
+            if (!$uri)
+            {
+                $this->redirect('/p/' . $page->id . '/' . $page->uri);
+            }
+
+            $page->childrens  = Model_Page::getChildrenPagesByParent($page->id);
+
+            if ($this->user->isAdmin || $this->user->id == $page->author->id){
+                $this->view['can_modify_this_page'] = true;
+            }else{
+                $this->view['can_modify_this_page'] = false;
+            }

```

Comment:
$this->view['can_modify_this_page'] = $this->user->isAdmin || $this->user->id == $page->author->id;